src/clj_3df/compiler.cljc

```diff
@@ -32,6 +32,16 @@
 
 (def ^{:arglists '([pred] [pred coll])} separate (juxt filter remove))
 
+(defn remove-once [args coll]
+  (let [pred (into #{} args)]
+    ((fn inner [coll pred]
```

Comment:
This is unneccessarily confusing, please just use an extra let binding for this function.